algorithms/aflplusplus/aflplusplus_mutation_hierarflow_routines.cpp

```diff
@@ -50,4 +50,71 @@ AFLMutCalleeRef<AFLplusplusState> HavocTemplate<AFLplusplusState>::operator()(
   return this->GoToDefaultNext();
 }
 
+template <>
+AFLMutCalleeRef<AFLplusplusState>
+SplicingTemplate<AFLplusplusState>::operator()(
+    AFLMutatorTemplate<AFLplusplusState>& mutator) {
+  // Declare the alias just to omit "this->" in this function.
+  auto& state = this->state;
+
+  if (!state.use_splicing || state.setting->ignore_finds) {
+    return this->GoToDefaultNext();
+  }
+
+  u32 splice_cycle = 0;
+  while (splice_cycle++ < option::GetSpliceCycles(state) &&
+         state.queued_paths > 1 && mutator.GetSource().GetLen() > 1) {
+    /* Pick a random queue entry and seek to it. Don't splice with yourself. */
+
+    u32 tid;
+    do {
+      using afl::util::UR;
+      tid = UR(state.queued_paths, state.rand_fd);
+    } while (tid == state.current_entry);
+
+    /* Make sure that the target has a reasonable length. */
+
+    while (tid < state.case_queue.size()) {
+      if (state.case_queue[tid]->input->GetLen() >= 2 &&
+          tid != state.current_entry)
+        break;
+
+      ++tid;
+    }
+
+    if (tid == state.case_queue.size()) continue;
+
+    auto& target_case = *state.case_queue[tid];
+    state.splicing_with = tid;
+
+    /* Read the testcase into a new buffer. */
+
+    target_case.input->Load();
+    bool success = mutator.Splice(*target_case.input);
+    target_case.input->Unload();
+
+    if (!success) {
+      continue;
+    }
+
+    using afl::dictionary::AFLDictData;
+
+    if (this->DoHavoc(mutator, *state.mutop_optimizer,
+                      aflplusplus::havoc::AFLplusplusCustomCases,
+                      utils::StrPrintf("more_splice %u", splice_cycle),
+                      "more_splice", state.orig_perf,
+                      option::GetSpliceHavoc(state), option::STAGE_SPLICE)) {
```

Comment:
The content of this explicit specialization is almost the same as AFL's `SplicingTemplate<State>` (i.e. the default implementation).
The diff lies in this part (calling `AFLplusplusCusomCases`).